### PR TITLE
Fix: should keep selected language tab position in search screen

### PR DESCRIPTION
**Pull request by cooltey**
If you try to search an article in a different language, the tab position should keep on the selected language after returning from the article.

Also, if you tap a language from the "Wikipedia languages" screen, it should be automatically selected (and scroll to it) in the search screen.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/views/LanguageScrollView.java [hunk 1]
```diff
@@ -101,27 +97,34 @@ private void updateTabView(boolean selected, TabLayout.Tab tab) {
...
         for (int i = 0; i < languageCodes.size(); i++) {
             TabLayout.Tab tab = horizontalLanguageScroll.newTab();
             tab.setCustomView(createLanguageTab(languageCodes.get(i)));
             horizontalLanguageScroll.addTab(tab);
             updateTabView(false, tab);
         }
-        selectLanguageTab(position);
+
+        if (horizontalLanguageScroll != null && horizontalLanguageScroll.getTabAt(position) != null) {
+            horizontalLanguageScroll.postDelayed(() -> horizontalLanguageScroll.getTabAt(position).select(), 100);
```
[the remark this answers is not in the document]
Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/LanguageScrollView.java]: Done!

**Comment by cooltey** [comment]
> It might be a lot less complicated, and even more useful, if we just save the last-used language position in a `Pref` (i.e. save it inside `onPause()`), and then restore it in `onStart()` where we currently set it to `0`.

Makes sense. Added!

File: app/src/main/java/org/wikipedia/views/LanguageScrollView.java [hunk 2]
```diff
@@ -101,27 +97,39 @@ private void updateTabView(boolean selected, TabLayout.Tab tab) {
...
         for (int i = 0; i < languageCodes.size(); i++) {
             TabLayout.Tab tab = horizontalLanguageScroll.newTab();
             tab.setCustomView(createLanguageTab(languageCodes.get(i)));
             horizontalLanguageScroll.addTab(tab);
             updateTabView(false, tab);
         }
-        selectLanguageTab(position);
+
+        if (horizontalLanguageScroll != null && horizontalLanguageScroll.getTabAt(position) != null) {
+            horizontalLanguageScroll.postDelayed(() -> {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/views/LanguageScrollView.java:121]: What's the reason for `post()`ing this?  It should work fine without a post.

Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/LanguageScrollView.java:121]: It fixes the issue that the scroll view does not scroll to the selected item after tapping on the last item in the Wikipedia languages screen(the item can not be seen by default).

Add ~5 languages and make sure the last item is not able to see -> when you tapping on the "more" to go to the "Wikipedia languages" -> tap on the last item -> it should scroll to the last item.

Comment (dbrant) [reply app/src/main/java/org/wikipedia/views/LanguageScrollView.java:121]: OK, in that case is there a reason for the 100ms delay? Why not just `post` without delay?

Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/LanguageScrollView.java]: Thanks, you're correct. Done!

**Pull request merged**